2kwebserver_merge_to_mega

s0170071 opened the pull request:
as requested, 2kwebserver just merged into mega. 
Compiles with Arduino IDE and runs fine on ESP.

s0170071 committed:
Webserver min memory

memory went as low as 3k, so wait for it to return ...

s0170071 committed:
Adjusted to min 7k free

s0170071 committed:
Timeout 100ms

s0170071 committed:
tcp_cleanup for LWIP1 & 2

s0170071 committed:
Adapted timeout

s0170071 committed:
trigger travis re-compile

trigger travis re-compile to see if the compile failure was due to a quick re-submission.

s0170071 committed:
try fix travis

s0170071 committed:
Merge remote-tracking branch 's0170071/webserverminmem' into webserverminmem

s0170071 committed:
Does Travis like extern "C" ?

s0170071 committed:
Wait for memory be like before send chunked

goes not below 9680 bytes now

s0170071 committed:
"Content-Type","text/html"

s0170071 committed:
Optimized memory usage

memory never goes below 10k now. 
Added wait for memory return on header send, 
Added wait for memory return on chunk end send,
removed duplicate of data stashing with  getWebPageTemplateVar

s0170071 committed:
Merge remote-tracking branch 's0170071/webserverminmem' into webserverminmem

s0170071 committed:
2k Webserver

s0170071 committed:
Merge remote-tracking branch 's0170071/webserverminmem' into webserverminmem

s0170071 committed:
Merge branch 'webserverminmem' into mega

s0170071 committed:
fix travis

s0170071 committed:
fix travis

unused variable

s0170071 committed:
Unused variable

s0170071 committed:
navMenuIndex

fixed order. Needs to be set before stream starts.

s0170071 committed:
fix download page

don't stream this one.

s0170071 committed:
Add Ram usage debug message

usually shows: 
Ram usage: Webserver only: 1048 including Core: 3120

s0170071 committed:
fix Travis

uninitialized variable.

s0170071 committed:
Travis...

s0170071 committed:
Cleanup and +1k Ram

- on 2.4.0 git /low mem LWIP ram goes never below 20k, during TX core dives down to 18,7k (each main menu clicked once, plain new config.)
- Fixed json output
- cleanup of send function#
- un-split css, transfer to buffer byte by byte (prevents 1.7k temporary string)
- removed commented-out function

s0170071 committed:
Merge branch 'mega' into 2kRamWebserver

s0170071 committed:
4->12 Devices per page

that was easy :-)

s0170071 committed:
Fix travis

unsigned int -> int comparison

s0170071 committed:
Update .gitignore

s0170071 committed:
Merge remote-tracking branch 'origin/mega' into 2kRamWebserver

s0170071 committed:
Merge Webserver

s0170071 committed:
Fixed String redeclaration

s0170071 committed:
Merge remote-tracking branch 'origin/mega' into 2kwebservermergetomega

s0170071 committed:
Merge into mega

... and removed obsolete function

TD-er commented:
Can you also create a sub-branch based on the current head of v2.0, where these 3 file changes are included?
ESPEasy.ino and Misc.ino are best edited manual I guess and I think (you can probably make a better educated guess) all of Webserver.ino should be replaced into the v2.0?

Then this new branch should get its own PR and then we can test it over the weekend and perhaps then merge when all works out fine?
The versions based on the Mega branch run just fine on my nodes at home.

s0170071 commented:
sorry, cant port to v20. Pull request is a mess. Shows 83 contributors and several thousand lines of change. Got no clue where to start. I would definately mess up things...

TD-er commented:
OK, then I will create one.
Not sure if I have time this evening.
Maybe @psy0rz can create a test branch for v2.0?

s0170071 committed:
Merge branch 'mega' into 2kwebservermergetomega

s0170071 committed:
Merge branch 'mega' into 2kwebservermergetomega

s0170071 committed:
Merge remote-tracking branch 's0170071/2kws_dynpayload' into 2kws_dynpayload

s0170071 committed:
Merge branch '2kws_dynpayload' into 2kwebservermergetomega

s0170071 committed:
Merge branch '2kws_dynpayload' into 2kwebservermergetomega

s0170071 committed:
Merge remote-tracking branch 's0170071/2kwebservermergetomega' into 2kwebservermergetomega

s0170071 committed:
Merge remote-tracking branch 's0170071/2kwebservermergetomega' into 2kwebservermergetomega

s0170071 committed:
variable timeout

s0170071 committed:
Merge remote-tracking branch 's0170071/2kwebservermergetomega' into 2kwebservermergetomega

s0170071 committed:
Merge remote-tracking branch 's0170071/2kwebservermergetomega' into 2kwebservermergetomega

s0170071 committed:
fix travis

s0170071 commented:
I made that waiting for the memory return dependent on the available memory. If there is enough - do not wait. This results in +30% speed. Hardware page loads in 230-250ms now.

s0170071 committed:
Removed double entry on info page

s0170071 committed:
Fix compile error when SD card is enabled

TD-er commented:
I see you're adding a lot of changes/updates. When is this stable enough to create a testbuild for people to test?
When you continue to add commits to this branch, the pull request will be updated.

s0170071 commented:
It is stable now for two weeks. What I changed were only minor changes, nothing spectacular. Its: 

- modified timeout condition when TX with lots of memory
- double entry in the info page (probably happened when merging)
- compile error when SD is enabled (was behind an #ifdef) 

I fix errors while I spot them so you only have one PR to worry about. It usually results in a number of commits because Travis often finds something my IDE didn't.

I would pull it right to mega. Isn't that whats the dev branch is for ?

TD-er commented:
Yep it is, but strictly speaking, if I test it, consider it OK to merge and then another commit is pushed, it is not what I tested.

So if you're still working on some minor issues, I can wait for an hour or two ;)

s0170071 commented:
Its working as far as I can tell. Found no other issues. 
If something comes up we can always fix it later.

TD-er commented:
Agree, I am now creating a merge for v2.0 (actually a new commit + PR) and have to strip a few new features introduced in Mega in the last 4 months.

TD-er commented:
I created a merge, see #934
Included is also a link to the builds for the v2.0 branch to test.
